### PR TITLE
http-client-java, upgrade to typespec compiler 0.61

### PR DESCRIPTION
**Pull request by haolingdong-msft**

### CODE REVIEW

File: packages/http-client-java/emitter/src/code-model-builder.ts [hunk 1]
```diff
@@ -2111,7 +2111,7 @@ export class CodeModelBuilder {
           name: "string",
           crossLanguageDefinitionId: type.crossLanguageDefinitionId,
         },
-        description: type.doc,
```
Comment (haolingdong-msft) [inline packages/http-client-java/emitter/src/code-model-builder.ts:2114]: description is removed in this release.

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/property/additionalproperties/models/DifferentSpreadModelArrayRecord.java [hunk 2]
```diff
@@ -27,7 +27,7 @@ public class DifferentSpreadModelArrayRecord implements JsonSerializable<Differe
     private final String knownProp;
 
     /*
-     * Additional properties
+     * The model spread Record<ModelForRecord[]> with the different known property type
```
Comment (XiaofeiCao) [inline packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/property/additionalproperties/models/DifferentSpreadModelArrayRecord.java:30]: This javadoc looks a bit ugly to me... Where does it come from?

Comment (haolingdong-msft) [reply packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/property/additionalproperties/models/DifferentSpreadModelArrayRecord.java:30]: It's due to this change. https://github.com/microsoft/typespec/pull/4671#discussion_r1794758615

Comment (haolingdong-msft) [reply packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/property/additionalproperties/models/DifferentSpreadModelArrayRecord.java:30]: I can remove the line that sets `sdkDictType`'s `doc`, so there will be no generation diff. I also feel the javadoc is a bit ugly. What do you all think? @weidongxu-microsoft  @XiaofeiCao

**Review by XiaofeiCao — APPROVED** [review]

File: packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/property/additionalproperties/models/DifferentSpreadModelArrayRecord.java (hunk 2 above)
Comment (haolingdong-msft) [reply packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/property/additionalproperties/models/DifferentSpreadModelArrayRecord.java:30]: After discussing offline, we will leave it for now, and we can change if there is ask/complains later.

**Pull request merged**